tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -3279,6 +3325,8 @@ def __init__(self, init_args, init_func, next_func, finalize_func):
       finalize_func: A TensorFlow function that will be called on the result of
         `init_func` immediately before a C++ iterator over this dataset is
         destroyed. The return value is ignored.
+      output_spec: A nested structure of `tf.TypeSpec` objects describing the
```

Comment:
Would you expect this to ever be different from next_func.output_structure?

Reply:
Yes. From what I've read from code and checked, the next_func.output_structure contains the description of the data in the low-level terms of flat tensor list (what we get from the IteratorGetNext), while output_spec is this higher-lever form to which we want the data to be transformed after being passed back to us from ops level (and the transformation happens in OwnedIterator._next_internal).